front-end/src/state/requests.js

```diff
@@ -0,0 +1,39 @@
+import { productAction } from './contexts/actions';
+import { categoriesApi, productsApi } from './contexts';
+import axios from 'axios';
+
+export const fetchCategories = async (dispatch) => {
+	try {
+		const {
+			data: { categories },
+			status
+		} = await axios.get(categoriesApi.GET.getAllCategories);
+
+		if (status === 200) {
+			dispatch({ type: productAction.fetchCategories, payload: categories });
+		} else {
+			throw new Error('Server not responded');
+		}
+	} catch (error) {
+		console.log(error);
+		throw error;
+	}
+};
+
+export const fetchProducts = async (dispatch) => {
+	try {
+		const {
+			data: { products },
```

Comment:
i guess if we fetch both products and categories inside a single try block it would behave like parallel calling, don't remember it fully though.

Reply:
Nice suggestion it should work.